core/src/main/java/hudson/ExtensionListListener.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson;
+
+/**
+ * {@link ExtensionList} listener.
+ *
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+public abstract class ExtensionListListener {
+
+    /**
+     * {@link ExtensionList} has been refreshed.
+     * <p>
+     * This would be called when a dynamically loaded plugin introduces a new
+     * {@link ExtensionPoint} implementation that adds an entry to the {@link ExtensionList}
```

Comment:
Ahem! Template deleting? Cough cough! (Stephenc masquerading as success kid)

Sent from my iPhone

> On 26 Apr 2015, at 10:35, Tom Fennelly <notifications@github.com> wrote:
> 
> In core/src/main/java/hudson/ExtensionListListener.java:
> 
> > + * THE SOFTWARE.
> > + */
> > +package hudson;
> > +
> > +/**
> > + * {@link ExtensionList} listener.
> > + *
> > + * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
> > + */
> > +public abstract class ExtensionListListener {
> > +
> > +    /**
> > +     * {@link ExtensionList} has been refreshed.
> > +     * <p>
> > +     * This would be called when a dynamically loaded plugin introduces a new
> > +     * {@link ExtensionPoint} implementation that adds an entry to the {@link ExtensionList}
> @stephenc I don't think that can happen in reality without restarting the Jenkins instance, can it? I know ExtensionList has a remove method, but I think that's really just for completeness wrt overriding List.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/jenkins/pull/1673/r29108261%40github.com.